plutus-core/executables/pir/Main.hs

```diff
@@ -1,106 +1,288 @@
+  <> metavar "PIR-FORMAT"
+  <> value TextualPir
+  <> showDefault
+  <> help ("Input format: " ++ pirFormatHelp))
+
+pPirOutputFormat :: Parser PirFormat
+pPirOutputFormat = option (maybeReader pirFormatReader)
+  (  long "of"
+  <> long "output-format"
+  <> metavar "PIR-FORMAT"
+  <> value TextualPir
+  <> showDefault
+  <> help ("Output format: " ++ pirFormatHelp))
+
+pPirOptimiseOptions :: Parser PirOptimiseOptions
+pPirOptimiseOptions = PirOptimiseOptions <$> input <*> pPirInputFormat <*> output <*> pPirOutputFormat <*> printmode
+
+pAnalyseOptions :: Parser AnalyseOptions
+pAnalyseOptions = AnalyseOptions <$> input <*> pPirInputFormat <*> output
+
+languageReader :: String -> Maybe Language
+languageReader =
+    \case
+         "plc"  -> Just PLC
+         "uplc" -> Just UPLC
+         _      -> Nothing
+
+pLanguage :: Parser Language
+pLanguage = option (maybeReader languageReader)
+  (  long "language"
+  <> short 'l'
+  <> metavar "LANGUAGE"
+  <> value UPLC
+  <> showDefault
+  <> help ("Target language: plc or uplc")
+  )
+
+pOptimise :: Parser Bool
+pOptimise = switch'
+            (  long "dont-optimise"
+            <> long "dont-optimize"
+            <> help ("Turn off optimisations")
+            )
+
+pJustTest :: Parser Bool
+pJustTest = switch ( long "test"
+                   <> help "Just report success or failure, don't produce an output file"
+                   )
+
+pCompileOptions :: Parser CompileOptions
+pCompileOptions = CompileOptions
+               <$> pLanguage
+               <*> pOptimise
+               <*> pJustTest
+               <*> input
+               <*> pPirInputFormat
+               <*> output
+               <*> outputformat
+               <*> printmode
+
+pPirOptions :: Parser Command
+pPirOptions = hsubparser $
+             command "analyse"
+                  (info (Analyse <$> pAnalyseOptions) $
+                   progDesc $
+                   "Given a PIR program in flat format, deserialise and analyse the program, " <>
+                   "looking for variables with the largest retained size.")
+           <> command "compile"
+                  (info (Compile <$> pCompileOptions) $
+                   progDesc $
+                   "Given a PIR program in flat format, deserialise it, " <>
+                   "and test if it can be successfully compiled to PLC.")
+           <> command "convert"
+                  (info (Convert <$> convertOpts)
+                   (progDesc $
+                    "Convert a program between various formats" <>
+                    "(only 'textual' (default) and 'flat-named' are available for PIR)."))
+           <> command "optimise"
+                  (info (Optimise <$> pPirOptimiseOptions)
+                   (progDesc "Run the PIR optimisation pipeline on the input."))
+           <> command "optimize"
+                  (info (Optimise <$> pPirOptimiseOptions)
+                   (progDesc "Run the PIR optimisation pipeline on the input."))
```

Comment:
this duplication could be removed but not a big issue

Reply:
duplication should be removed.